### PR TITLE
New Gasses

### PR DESCRIPTION
**Pull request by nevimer**
Testing, check PR contents for now

### CODE REVIEW

**Comment by TheGandalf85** [comment]
image
Everyone be vomiting in the medbay. Scrubbers be no scrubbing. Help.

**Comment by TheGandalf85** [comment]
image
Gas Filter no show new gas.

**Comment by BurgerLUA** [comment]
I don't know if I like this change. I think corpses should still emit miasma, but the thresholds for when miasma gives diseases should just be increased and the rate at which gas is produced should be decreased.

File: modular_zubbers/code/modules/scents/rot.dm [hunk 1]
```diff
@@ -12,13 +12,14 @@
 	if(!istype(T) || T.planetary_atmos || T.return_air().return_pressure() > (WARNING_HIGH_PRESSURE - 10))
 		return
 	var/datum/gas_mixture/stank = new
-	ADD_GAS(/datum/gas/miasma, stank.gases)
-	stank.gases[/datum/gas/miasma][MOLES] = ROT_MOLES
+	if(owner)
+		ADD_GAS(/datum/gas/scent/cadaverine, stank.gases)
+		stank.gases[/datum/gas/scent/cadaverine][MOLES] = ROT_MOLES
+	ADD_GAS(/datum/gas/scent/putrescine, stank.gases)
```
Comment (TheGandalf85) [inline modular_zubbers/code/modules/scents/rot.dm:18]: You sure we should be stanking both cadaverine and putrescine here?

**Comment by BrandonSmith6** [comment]
Thanks to this PR what was supposed to be a calm shift turned into vomiting for an hour straight. I hate it.

**Comment by TheGandalf85** [comment]
I'm not digging the gameplay with this. Basically any part of the station even remotely near where dead bodies are (aka monkeys.), they get this issue. I don't think body bags stop this either, so certain areas like the morgue will just endlessly get filled up. The very grossed out moodlet effectively stunlocks people too.

Also, this doesn't play nice with atmos. Very low moles in large areas tend to "keep around" and sometimes keeps affecting everyone similar to plasma. Make sure the standard setting for all atmos air alarms accept these new gasses.

Recently deceased bodies will also generate this gas enmasse since **Every organ in their bodies will rot.** The gas spreads almost as bad as dwarf fortress miasma. There's a reason the original miasma doesn't generate or spread this fast.

**Comment by nevimer** [comment]
its a test, go complain on discord if its not about code.

**Comment by nevimer** [comment]
this isn't even finished or even tested yet and yall are downvoting it. tell you what, im going to just merge commits instead of testmerging if things keep up.

File: modular_zubbers/code/modules/scents/rot.dm (hunk 1 above)
Comment (nevimer) [reply modular_zubbers/code/modules/scents/rot.dm:18]: it's intentional to follow up with making this a unique thing for the mortician later.

**Comment by StrangeWeirdKitten** [comment]
Two things:
- I think water vapor should neutralize it.
- Can we tie organ decay releasing gasses to only mobs with ckeys? When you deal with a ton of dead corpses, like xenobiology, it becomes extremely annoying.

**Comment by Mewchild** [comment]
> this isn't even finished or even tested yet and yall are downvoting it. tell you what, im going to just merge commits instead of testmerging if things keep up.

The reason it's being downvoted is because it's not a fun mechanic to deal with. From a gameplay point, what does this add? Functionally, it's just gas that causes disgust. With the amount of death that happens on this server (from mining, xenobio, medical, and antag antics), it rapidly devolves into a stationwide issue, every single round. What gameplay benefit, from a fun perspective, is there to be gained? What's the gameplay niche is being filled by this and the recent corpse changes that necessitate rotting give off undesirable gasses? From what I've experienced, the entire mechanic is unfun, and unavoidable.

**Comment by mcmeiler** [comment]
Can we please force people to actually write PR descriptions before TMing them, jesus christ.

## Not everyone can read code, or be bothered to read an entire PR's worth of changes.

**Comment by projectkepler-ru** [comment]
Documentation please? @nevimer

**Comment by BubbleTrroubles** [comment]
> this isn't even finished or even tested yet and yall are downvoting it. tell you what, im going to just merge commits instead of testmerging if things keep up.

Really now, is that what you are going to do?

**Comment by Majkl-J** [comment]
> this isn't even finished or even tested yet and yall are downvoting it. tell you what, im going to just merge commits instead of testmerging if things keep up.

If it isn't near finished it shouldn't be testmerged, especially with no PR description to help players know what it does.
Also threatening to commit straight to master is just scummy and really won't gain you people's approval

**Comment by sunofang** [comment]
The new gasses are pretty much pointless. Please tell us why you need 3 different gasses from corpses.

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]
Please, provide documentation on this, and, reduce the rate at which the gas is produced from the bodies,

As it stand, this can become an entire station wide threat by literally just dragging body around or strategically throwing rotting organs into place people don't often check, which, isn't really fun to deal with. (ergo, imagine having to wear internal all round or having to somehow forego breathing. Not an issue for people who play non-standard race, an issue for the rest of us.)

Lastly, please test this out locally before putting it on the server to ensure your PR is actually working as intended

**Comment by nevimer** [comment]
Your comment is based on a outdated information. Please see the commit log.

**Comment by StrangeWeirdKitten** [comment]
This PR is under discussion and there are changes planned after discussing it with Ozzo about it.

**Comment by Majkl-J** [comment]
Stale for a month, reopen when work continues

**Pull request closed without merging**